elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation;
+
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.DataStoreTransaction;
+
+/**
+ * A {@link QueryEngine} is an abstraction layer that {@link DataStoreTransaction} leverages to optimize query and send
```

Comment:
I would say something like:

```A QueryEngine is an abstraction that an AggregationDataStore leverages to run analytic queries (OLAP style) against an underlying persistence layer.```

Also - why Drill?   `There will be` implies that we plan on building these things.  I think the Diagram is fine, but I wouldn't imply future support for things we don't currently plan on building.  Maybe substitute Drill with Druid.   Also, I don't think we'll need both Hibernate and JPA.  In In Elide 5.0, the explicit hibernate dependency will be greatly diminished.